### PR TITLE
Load user prefs fast

### PR DESCRIPTION
**Pull request by eishay**
@ymatsuda @andrewconner @2is10

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/model/UserValueRepo.scala [hunk 1]
```diff
@@ -58,6 +59,16 @@ class UserValueRepoImpl @Inject() (
     }
   }
 
+  def getValues(userId: Id[User], names: String*)(implicit session: RSession): Map[String, Option[String]] =
+    valueCache.bulkGetOrElseOpt(names map { name => UserValueKey(userId, name)} toSet ) { missingNames =>
+      val missingValues = missingNames map {missingName =>
+        missingName -> (for(f <- rows if f.state === UserValueStates.ACTIVE && f.userId === userId && f.name === missingName.key) yield f.value).firstOption.map(_.value)
```
Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/model/UserValueRepo.scala:65]: can you just call `getUserValue` here?

File: kifi-backend/modules/common/app/com/keepit/model/UserValue.scala [hunk 2]
```diff
@@ -44,3 +43,27 @@ object Gender {
   }
 }
 
+object UserValues {
+
+  trait UserValueHandler[T] {
+    def name: String
+    def get(valOpt: Option[String]): T
```
Comment (2is10) [inline kifi-backend/modules/common/app/com/keepit/model/UserValue.scala:50]: `parse` might be a more helpful name

File: kifi-backend/modules/shoebox/app/com/keepit/model/UserValueRepo.scala (hunk 1 above)
Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/model/UserValueRepo.scala:65]: yes, i'll dry it out

**Comment by 2is10** [comment]
cool

File: kifi-backend/modules/common/app/com/keepit/model/UserValue.scala (hunk 2 above)
Comment (eishay) [inline kifi-backend/modules/common/app/com/keepit/model/UserValue.scala:50]: sure

**Comment by eishay** [comment]
closing again, i'll do more type safe refactoring so it may be significant.

**Pull request closed without merging**